docs/src/proposals/epoch_accounts_hash.md

```diff
@@ -0,0 +1,203 @@
+
+
+### Implementation
+
+Performing a full accounts hash takes a relatively long time to complete.  For
+this reason, the EAH calculation must take place in the background.
+
+In order for all the validators to calculate the same accounts hash, the
+calculation must be based on the same view of all accounts.  This means the EAH
+must be based on a predetermined slot.  This will be known as the `start slot`.
+The `start slot` is calculated as an offset from the first slot of an epoch.
+This offset will be known as the `start offset`. 
+
+Similarly, all the validators must save the EAH into a bank at a predetermined
+slot, and offset from the first slot of an epoch.  This will be known as the
+`stop slot` and `stop offset`, respectively.
+
+* The `start offset` will be set at one-quarter into the epoch.
+* The `stop offset` will be set at three-quarters into the epoch.
+* For epochs with 432,000 slots, the `start offset` will be 108,00 and the
+  `stop offset` will be 324,000.
+
+These constants may be changed in the future, or may be determined at runtime.
+The main justifications for these values are:
+1. Do not start the EAH calculation at the beginning of an epoch, as the
+   beginning of an epoch is already a time of contention and stress.  There is
+   no reason to make this worse.
+2. The bank to save the EAH into—the `stop offset`—should be sufficiently far
+   in the future to guarantee all validators are able to complete the accounts
+   hash calculation in time.
+3. The EAH window, `start offset` to `stop offset`, should be small.  The
+   validator will need to keep around all the state for the accounts at the
+   `start slot` while the EAH calculation is occurring; this is expensive, and
+   that duration should be minimized.
+4. The `start offset` should be *after* the `rewarding interval`
+   (from [Partitioned Inflationary Rewards Distribution](https://github.com/solana-labs/solana/pull/27455)).
+
+Once the EAH calculation is complete, it must be saved somewhere.  Since this
+occurs in the background, there is not an associated `Bank` that would make
+sense to save into.  Instead, a new field will be added to `AccountsDb` that
+will store the EAH.  Later, the bank at slot `stop slot` will read the EAH from
+`AccountsDb` and hash it into its own hash (aka _bank hash_).
+
+EAH calculation will use the existing _accounts background services_ (_ABS_) to
+perform the actual calculation.  Requests for EAH calculation will be sent from
+`bank_forks::set_root()`, with a new request type to distinguish an EAH request
+from a Snapshot request.  Since the EAH will be part of consensus, it is not
+optional; EAH requests will have the highest priority in ABS, and will be
+processed first/instead of other requests.
+
+
+#### Details
+
+#### Snapshots
+
+A snapshot contains all the state necessarily to reconstruct the cluster as of
+a certain slot.  A snapshot taken early in the epoch will need to contain the
```

Comment:
instead of 'early in the epoch' should this be 'after than the start offset' or something. For examplesnapshots at slot 2 in the epoch do not need to do anything about eah.